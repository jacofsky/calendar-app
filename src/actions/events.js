import { types } from "../types/types";
import { fetchConToken } from "../helpers/fetch";
import Swal from "sweetalert2";
import { prepareEvents } from "../helpers/prepareEvents";

export const eventStartAddNew = (event) => {
    return async(dispatch, getState) => {

        const {uid, name} = getState().auth

        try {
            const resp = await fetchConToken('events', event, 'POST')
            const body = await resp.json()

            if (body.ok) {

                event.id = body.evento.id
                event.user = {
                    _id: uid,
                    name: name
                }
                dispatch(eventAddNew(body.evento))
            }
            
        } catch (error) {
            console.log('error')
            Swal.fire('Error', 'Ah surgido un problema misterioso! ðŸ˜°', 'error')
        }
    }
}

export const eventStartLoading = () => {
    return async(dispatch) =>{

        try {
            const resp = await fetchConToken('events')
            const body = await resp.json()

            if (body.ok) {
                const events = (prepareEvents(body.msg))
                dispatch(eventsLaoded(events))
            }

        } catch (error) {
            console.log('error')
            Swal.fire('Error', 'Ah surgido un problema misterioso! ðŸ˜°', 'error')
        }
    }
}

export const eventStartUpdate = (event) => {
    return async(dispatch) => {

        try {
            const resp = await fetchConToken(`events/${event.id}`, event, 'PUT')
            const body = await resp.json()

            console.log(body)

            if (body.ok) {
                dispatch(eventUpdated(event))
            } else {
                Swal.fire('Error', body.msg, 'error')
            }

        } catch (error) {
            console.log('error')
        }
    }
}

export const eventStartDelete = () => {
    return async(dispatch, getState) => {

        const  {id} = getState().calendar.activeEvent

        try {
            const resp = await fetchConToken(`events/${id}`, {}, 'DELETE')
            const body = await resp.json()

            console.log(body)

            if (body.ok) {
                dispatch(eventDeleted())
            } else {
                Swal.fire('Error', body.msg, 'error')
            }

        } catch (error) {
            console.log('error')
        }
    }
}

export const eventLogout = () => ({
    type: types.eventLogout
})

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event
})

export const eventClearActiveEvent = () => ({
    type: types.eventClearActiveEvent
})

const eventDeleted = () => ({
    type: types.eventDeleted
})

const eventUpdated = (event) => ({
    type: types.eventUpdated,
    payload: event
})

const eventsLaoded = (eventos) => ({
    type: types.eventLoaded,
    payload: eventos
})

const eventAddNew = (event) => ({
    type: types.eventAddNew,
    payload: event
})

