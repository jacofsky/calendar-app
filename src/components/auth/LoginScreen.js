import React from 'react';
import './login.css';

import { useForm } from '../../hooks/useForm';
import { useDispatch } from 'react-redux';
import { startLogin, startRegister } from '../../actions/auth';
import Swal from 'sweetalert2';

const LoginScreen = () => {

    const [ formLoginValues, handleLoginInputChange] = useForm( {
        lEmail: '',
        lPassword: ''
    });

    const {lEmail, lPassword} = formLoginValues

    const [ formRegisterValues, handleRegisterInputChange] = useForm( {
        rNombre: '',
        rEmail: '',
        rPassword1: '',
        rPassword2: ''
    });

    const {rNombre, rEmail, rPassword1, rPassword2} = formRegisterValues

    const dispatch = useDispatch()

    const handleLogin = (e) => {
        e.preventDefault()
        dispatch(startLogin(lEmail, lPassword))
    }

    const handleRegister = (e) => {
        e.preventDefault()

        if(rPassword1 !== rPassword2) {
            return Swal.fire('Error', 'Las constrase単as deben ser iguales', 'error')
        } else{
            dispatch(startRegister(rNombre, rEmail, rPassword1))
        } 
        
    }

    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-1">
                    <h3 className="fs-1">Ingreso</h3>
                    <form onSubmit={handleLogin}>
                        <div className="form-group">
                            <input 
                                type="text"
                                className="form-control"
                                placeholder="Correo"
                                name='lEmail'
                                onChange={handleLoginInputChange}
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contrase単a"
                                name='lPassword'
                                onChange={handleLoginInputChange}
                            />
                        </div>
                        <div className="form-group">
                            <input 
                                type="submit"
                                className="btnSubmit"
                                value="Login" 
                            />
                        </div>
                    </form>
                </div>

                <div className="col-md-6 login-form-2">
                    <h3 className="fs-1">Registro</h3>
                    <form onSubmit={handleRegister}>
                        <div className="form-group">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Nombre"
                                name='rNombre'
                                onChange={handleRegisterInputChange}
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Correo"
                                name='rEmail'
                                onChange={handleRegisterInputChange}
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contrase単a" 
                                name='rPassword1'
                                onChange={handleRegisterInputChange}
                            />
                        </div>

                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repita la contrase単a" 
                                name='rPassword2'
                                onChange={handleRegisterInputChange}
                            />
                        </div>

                        <div className="form-group">
                            <input 
                                type="submit" 
                                className="btnSubmit" 
                                value="Crear cuenta" />
                        </div>
                    </form>
                </div>
            </div>
        </div>
    )
}

export default LoginScreen