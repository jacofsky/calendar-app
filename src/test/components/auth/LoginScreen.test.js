import React from 'react'
import LoginScreen from '../../../components/auth/LoginScreen';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import { startLogin, startRegister } from '../../../actions/auth';
import Swal from 'sweetalert2';

jest.mock(('../../../actions/auth'), () => ({
    startLogin: jest.fn(),
    startRegister: jest.fn()
}))



jest.mock(('sweetalert2'), () => ({
    fire: jest.fn()
}))


const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {}

let store = mockStore(initState)
store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={store}>
        <LoginScreen/>
    </Provider>
)


describe('Test in LoginScreen', () => { 

    beforeEach(() => {
        jest.clearAllMocks()
    })

    test('should correctly', () => { 
        expect(wrapper).toMatchSnapshot()
    })

    test('should call dispatch login', () => { 
        wrapper.find('input[name="lEmail"]').simulate('change', {
            target: {
                name: 'lEmail',
                value: 'jaco@gmail.com'
            }
        })
        wrapper.find('input[name="lPassword"]').simulate('change', {
            target: {
                name: 'lPassword',
                value: '123456'
            }
        })

        wrapper.find('form').at(0).prop('onSubmit')({
            preventDefault(){}
        })

        expect(startLogin).toHaveBeenCalledWith('jaco@gmail.com', '123456')

    })

    test('should not register if the passwords are not equal', () => { 

        wrapper.find('input[name="rNombre"]').simulate('change', {
            target: {
                name: 'rNombre',
                value: 'Jaco'
            }
        })
        wrapper.find('input[name="rEmail"]').simulate('change', {
            target: {
                name: 'rEmail',
                value: 'jaco@gmail.com'
            }
        })
        wrapper.find('input[name="rPassword1"]').simulate('change', {
            target: {
                name: 'rPassword1',
                value: '123456'
            }
        })

        wrapper.find('input[name="rPassword2"]').simulate('change', {
            target: {
                name: 'rPassword2',
                value: '1234567'
            }
        })

        wrapper.find('form').at(1).prop('onSubmit')({
            preventDefault(){}
        })

        expect(startRegister).not.toHaveBeenCalled()
        expect(Swal.fire).toHaveBeenCalled()

    })

    test('should not register if the passwords are not equal', () => { 

        wrapper.find('input[name="rNombre"]').simulate('change', {
            target: {
                name: 'rNombre',
                value: 'Jaaaaco'
            }
        })
        wrapper.find('input[name="rEmail"]').simulate('change', {
            target: {
                name: 'rEmail',
                value: 'jaco@gmail.com'
            }
        })
        wrapper.find('input[name="rPassword1"]').simulate('change', {
            target: {
                name: 'rPassword1',
                value: '123456'
            }
        })

        wrapper.find('input[name="rPassword2"]').simulate('change', {
            target: {
                name: 'rPassword2',
                value: '123456'
            }
        })

        wrapper.find('form').at(1).prop('onSubmit')({
            preventDefault(){}
        })

        expect(Swal.fire).not.toHaveBeenCalled()
        expect(startRegister).toHaveBeenCalledWith("Jaaaaco", "jaco@gmail.com", "123456")

    })

 })