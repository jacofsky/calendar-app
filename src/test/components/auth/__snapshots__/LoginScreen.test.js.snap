// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in LoginScreen should correctly 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3
        className="fs-1"
      >
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="lEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="lPassword"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
    <div
      className="col-md-6 login-form-2"
    >
      <h3
        className="fs-1"
      >
        Registro
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rNombre"
            onChange={[Function]}
            placeholder="Nombre"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rPassword1"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rPassword2"
            onChange={[Function]}
            placeholder="Repita la contraseña"
            type="password"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Crear cuenta"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
